language/move-lang/src/lib.rs

```diff
@@ -180,6 +180,41 @@ pub fn move_continue_up_to(pass: PassResult, until: Pass) -> Result<PassResult,
     run(pass, until)
 }
 
+/// Similar to `move_compile`, but with an additional `allow_shadow` flag to indicate that modules
+/// defined in targets can shadow (i.e., override) the modules in dependencies. In other words,
+/// if the same module id (<address::name>) is found in both targets and dependencies, the module
+/// definition in targets takes priority.
+///
+/// This function also allows to specify a stopping point of the compilation process. Similar to
+/// `move_continue_up_to`, the stopping point is inclusive, meaning that the pass specified by
+/// `until: Pass` will be run
+pub fn move_compile_up_to(
+    targets: &[String],
+    deps: &[String],
+    sender_opt: Option<Address>,
+    interface_files_dir_opt: Option<String>,
+    allow_shadow: bool,
+    until: Pass,
+) -> anyhow::Result<(FilesSourceText, Result<PassResult, Errors>)> {
```

Comment:
Maybe a larger question, but I'm wondering why we are using two different versions of `Result` here...It seems like other places in this file are doing a similar thing, so this doesn't block this PR, but it might be useful to mark this as a future cleanup unless there's a good reason for this. @tnowacki curious if you think this cleanup would make sense.

Reply:
In fact, I had the same wondering here :)

Reply:
Its tricky....

The outer result is for random IO nonsense, the inner one is for errors from the compiler itself. 
IF you wanted to clean this up, you could handroll an enum for the 3 states. It would be clearer but I'm not sure if it would be simpler